### DISCUSSION TITLE
ADFS integration with Gitlab using AD groups

### DISCUSSION THREAD

**Discussion by sagarkulkarni1989** [community/community#55270] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Show & Tell

### Body

We are using Gitlab community Edition and our requirement is to intgrate ADFS with Gitlab using AD group. Could you please help on this.
As of now we have created AD group to test integration with gitlab with AD groups
Unable to see those groups in invite group option.

Could you please help out here

**Comment by JesusCastroFernandez** [comment]
Para integrar ADFS con GitLab usando grupos de AD, primero es necesario configurar la autenticación SAML en GitLab y luego configurar la integración con ADFS. Aquí hay una guía paso a paso para hacerlo:

Configurar la autenticación SAML en GitLab
En GitLab, vaya a "Configuración del sistema" y seleccione "Autenticación de terceros" en la barra lateral izquierda.
Habilite la autenticación SAML y configure los siguientes parámetros:
Proveedor de identidad SAML: seleccione "Proveedor de identidad personalizado".
URL de inicio de sesión SAML: la URL de inicio de sesión de ADFS.
Certificado SAML: el certificado SAML de ADFS (debe ser una cadena Base64).
URL de retorno de llamada SAML: la URL de retorno de llamada de GitLab (por ejemplo, https://gitlab.example.com/users/auth/saml/callback).
Configuración SAML personalizada: aquí es donde se especifican los mapeos de atributos SAML a campos de usuario en GitLab. Puede utilizar el siguiente ejemplo:
json
Copy code
{
    "assertion_consumer_service_url": "https://gitlab.example.com/users/auth/saml/callback",
    "issuer": "https://adfs.example.com/adfs/services/trust",
    "idp_sso_target_url": "https://adfs.example.com/adfs/ls/",
    "idp_cert_fingerprint": "AA:BB:CC:DD:EE:FF:...",
    "name_identifier_format": "urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified",
    "attribute_statements": {
        "username": ["http://schemas.xmlsoap.org/ws/2005/05/identity/claims/name"],
        "email": ["http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress"],
        "first_name": ["http://schemas.xmlsoap.org/ws/2005/05/identity/claims/givenname"],
        "last_name": ["http://schemas.xmlsoap.org/ws/2005/05/identity/claims/surname"],
        "groups": ["http://schemas.microsoft.com/ws/2008/06/identity/claims/role"]
    }
}
Este ejemplo mapea los atributos SAML "username", "email", "first_name" y "last_name" a los campos de usuario correspondientes en GitLab, y mapea el atributo SAML "groups" a los grupos de usuario en GitLab.

Configurar la integración con ADFS
En ADFS, agregue una nueva aplicación Relying Party Trust y configure los siguientes parámetros:
URL de inicio de sesión: la URL de inicio de sesión SAML de GitLab.
Certificado de firma: el certificado SAML de GitLab (debe ser una cadena Base64).
Configuración de atributos de reclamo: aquí es donde se especifican los mapeos de atributos de usuario de AD a atributos SAML. Puede utilizar el siguiente ejemplo:
bash
Copy code
username -> NameID
email -> http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress
first_name -> http://schemas.xmlsoap.org/ws/2005/05/identity/claims/givenname
last_name -> http://schemas.xmlsoap.org/ws/2005/05/identity/claims/surname
groups -> http://schemas.microsoft.com/ws/2008/06/identity

**Comment by ghostinhershell** [comment]
Hey there!  👋

Thanks for posting in the GitHub Community, @sagarkulkarni1989! We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category. The Accessibility category is a place for our community to discuss and provide feedback on the digital accessibility of GitHub products. Digital accessibility means that GitHub tools, and technologies, are designed and developed so that people with disabilities can use them.

I've gone ahead and moved this to the correct category for you. However, I do want to note that this is the GitHub Community on Discussions. You may have more luck finding an answer to your question by reaching out to either the GitLab's community or reaching out to GitLab's support team. 

Good luck!
